File: backend/Model/Shipment.cs
namespace CosmoCargo.Model
{
    public class Shipment
    {
        public Guid Id { get; set; }
        public Guid CustomerId { get; set; }
        public Guid? PilotId { get; set; }
        public ShipmentContact Sender { get; set; } = new();
        public ShipmentContact Receiver { get; set; } = new();
        
        // Package information
        public decimal Weight { get; set; }
        public string Category { get; set; } = string.Empty;
        public string Priority { get; set; } = string.Empty;
        public string? Description { get; set; }
        public bool HasInsurance { get; set; }
        
        // Status and tracking
        public ShipmentStatus Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Customs
        public CustomsForm Customs { get; set; } = new();

        // Navigation properties
        public virtual User Customer { get; set; } = null!;
        public virtual User? Pilot { get; set; }
    }
}

File: frontend/src/model/shipment.ts
import { ShipmentStatus } from './types';
import User from './user';

export interface ShipmentContact {
    name: string;
    email: string;
    planet: string;
    station: string;
}

export interface CustomsForm {
    containsLifeforms: boolean;
    lifeformType?: string | null;
    isPlasmaActive: boolean;
    plasmaStabilityLevel?: number | null;
    originPlanetLawsConfirmed: boolean;
    customsNotes?: string | null;
    quarantineRequired?: boolean | null;
}

export default interface Shipment {
    id: string;
    customerId: string;
    pilotId?: string | null;
    sender: ShipmentContact;
    receiver: ShipmentContact;
    weight: number;
    category: string;
    priority: string;
    description?: string | null;
    hasInsurance: boolean;
    status: ShipmentStatus;
    createdAt: string;
    updatedAt: string;
    pilot?: User | null;
    customs?: CustomsForm;
}
